torchvision/csrc/io/image/cpu/decode_jpeg.cpp

```diff
@@ -24,19 +24,15 @@ struct torch_jpeg_mgr {
 static void torch_jpeg_init_source(j_decompress_ptr cinfo) {}
 
 static boolean torch_jpeg_fill_input_buffer(j_decompress_ptr cinfo) {
-  torch_jpeg_mgr* src = (torch_jpeg_mgr*)cinfo->src;
   // No more data.  Probably an incomplete image;  Raise exception.
   torch_jpeg_error_ptr myerr = (torch_jpeg_error_ptr)cinfo->err;
   strcpy(myerr->jpegLastErrorMsg, "Image is incomplete or truncated");
   longjmp(myerr->setjmp_buffer, 1);
-  src->pub.next_input_byte = EOI_BUFFER;
-  src->pub.bytes_in_buffer = 1;
-  return TRUE;
```

Comment:
I'm curious as to why this is needed, do you remember what the compilation warning was?

Doesn't this potentially change the behavior?

Also, should the function be declared as `static void` now that the `return` is removed?

Reply:
The warning was about unused/unreachable code. Checkout [longjump](https://en.cppreference.com/w/cpp/utility/program/longjmp).

The method can't be declared as void as it must match the signature defined in `jpeg_source_mgr`.

Reply:
Thanks for fixing this!

@andfoy do we need to destroy any structures before the `longjmp`, or set those elements before the `longjmp`?